projects/cdk/utils/dom/get-selection-text.ts

```diff
@@ -0,0 +1,23 @@
+type Target = Element | EventTarget | null;
+
+/**
+ * @description:
+ * cross browser way to get selected text
+ *
+ * History:
+ * BUG - window.getSelection() fails when text selected in a form field
+ * https://bugzilla.mozilla.org/show_bug.cgi?id=85686
+ */
+export function tuiGetSelectionText(windowRef: Window, target?: Target): string | null {
+    const selection = windowRef.getSelection();
```

Comment:
How about we pass `document` instead so we only have one argument?
https://developer.mozilla.org/en-US/docs/Web/API/Document/getSelection
We can then use `activeElement` to test if we are in an input field:
```js
document.activeElement.value.substring(
  document.activeElement.selectionStart,
  document.activeElement.selectionEnd
)
```

Reply:
fixed